libfs/file.go

```diff
@@ -114,3 +118,65 @@ func (f *File) Close() error {
 	f.node = nil
 	return nil
 }
+
+func (f *File) getLockID() keybase1.LockID {
+	// We must not change or remove this since it'd break locking guarantees.
+	return keybase1.LockIDFromBytes(
+		bytes.Join([][]byte{
+			f.fs.GetLockNamespace().Bytes(),
+			[]byte(f.Name()),
```

Comment:
Also, using `f.Name()` will work fine in practice, but technically `f.fs` could be chroot'd to some subdirectory within the repo, and `f.Name()` might not be truly unique within the repo.  You could use `f.fs.Root()` to fix that, but it would have some extra components like tlf and repo name that you wouldn't need.

Reply:
>  in practice
Meaning we never chroot in `git` right?

In context of git, I'm concerned about including `f.fs.Root()` since repo name can change. I think we should only associate with the path inside the repo and the repo ID. But good point on chroot from billy perspective. But I don't have a better idea. Perhaps we can alter the LockNamespace in `fs.Chroot`, or just mention in the comment of Lock() that chroot breaks locking?

Reply:
> Meaning we never chroot in git right?

Not for the FS we pass into go-git, where the locking happens.

You could just strip off the first N (5?) elements from Root() if you want to get the path just within the repo.

Reply:
But then that's making the assumption in billy that we are using `git`, which is not ideal right?

Reply:
Hmm, I guess? I don't care much though.  In the custom context object you could have another method that transforms the filename maybe, and we could have a custom one for git, just like the namespace bytes?

Reply:
Since the locking namespace is set in `FS`, I think it's better to merge this into `FS`. I'm pushing a commit that just builds a unix canonical path and uses that as default namespace. In case of Chroot it appends the changed path to the original namespace. This should work for both default case and git. Let me know if it sounds good!